src/shared/condition.c

```diff
@@ -138,6 +140,49 @@ static int condition_test_kernel_command_line(Condition *c) {
         return false;
 }
 
+static int condition_test_user(Condition *c) {
+        uid_t id;
+        int r;
+        _cleanup_free_ const char * username = NULL;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_USER);
+
+        r = parse_uid(c->parameter, &id);
+        if (r >= 0)
+                return id == getuid() || id == geteuid();
+        else if (r != -EINVAL)
+                return r;
+
+        username = getusername_malloc();
+        if (!username)
+                return -ENOMEM;
+
+        return streq(username, c->parameter);
+}
+static int condition_test_group(Condition *c) {
+        gid_t id;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_GROUP);
+
+        r = parse_gid(c->parameter, &id);
+        if (r >= 0)
+                return in_gid(id);
+        else if (r != -EINVAL)
+                return r;
+
+        // Avoid any NSS lookups if we are PID1
+        if (getpid() == 1) {
+                return streq(c->parameter, "root");
+        }
+
+        return in_group(c->parameter);
```

Comment:
Hmm, I think we should probably eat up errors here. The problem is that resolving group names doesn't really always result in well-defined error numbers (as the error numbers are essentially up to the various NSS modules, and they generally don't agree much on which error to return when). Hence, given that this doesn't really matter much I'd just say that if the check can't be executed we'll consider it not holding, i.e.

```c
return in_group(c->parameter) > 0;
```